plugins/module_utils/azure_rm_common.py

```diff
@@ -675,19 +675,23 @@ def check_provisioning_state(self, azure_object, requested_state='present'):
                 self.fail("Error {0} has a provisioning state of {1}. Expecting state to be {2}.".format(
                     azure_object.name, azure_object.provisioning_state, AZURE_SUCCESS_STATE))
 
-    def get_blob_service_client(self, resource_group_name, storage_account_name):
+    def get_blob_service_client(self, resource_group_name, storage_account_name, auth_mode = 'key'):
```

Comment:
unexpected spaces, unexpected spaces around keyword / parameter equals

```suggestion
    def get_blob_service_client(self, resource_group_name, storage_account_name, auth_mode='key'):
```